### PR TITLE
Baleen Poet - Write Baleen Kotlin source code from Baleen types

### PR DESCRIPTION
**Pull request by kdallmeyer-sr**
Baleen Poet is an extension of the awesome [Kotlin Poet](https://github.com/square/kotlinpoet) library that supports writing Kotlin files. This library takes a BaleenType as input and then writes it to the file.  This is especially useful for Baleen generators where an external format like Json schema or Avro schemas can be imported to create a Baleen schema.

```kotlin
import com.shoprunner.baleen.describeAs
import com.shoprunner.baleen.poet.toFileSpec
import com.shoprunner.baleen.types.AllowsNull
import com.shoprunner.baleen.types.IntType
import com.shoprunner.baleen.types.StringType

val type = "Dog".describeAs("com.shoprunner.baleen.test") {
    "name".type(StringType())
    "numLegs".type(AllowsNull(IntType()))
}

type.toFileSpec().writeTo(File("src/"))
```
Creates src/com/shoprunner/baleen/test/Dog.kt
```kotlin
package com.shoprunner.baleen.poet.test

import com.shoprunner.baleen.Baleen.describe
import com.shoprunner.baleen.DataDescription
import com.shoprunner.baleen.types.AllowsNull
import com.shoprunner.baleen.types.IntType
import com.shoprunner.baleen.types.StringType

val Dog: DataDescription = describe("Dog", "com.shoprunner.baleen.poet.test", "") {
      it.attr(
        name = "name",
        type = StringType(min = 0, max = Int.MAX_VALUE)
      )
      it.attr(
        name = "numLegs",
        type = AllowsNull(IntType(min = Int.MIN_VALUE, max = Int.MAX_VALUE))
      )

    }
```

### CODE REVIEW

**Review by HawaiianSpork — APPROVED** [review]
Looks really good.  I love that you did this.  I'll let you squash-and-merge so you can continue to be the author.

File: baleen-poet/src/test/kotlin/com/shoprunner/baleen/poet/BaleenPoetTest.kt [hunk 1]
```diff
@@ -53,7 +53,7 @@ internal class BaleenPoetTest {
                 import com.shoprunner.baleen.BaleenType
                 import com.shoprunner.baleen.types.BooleanType
                 
-                val boolean: BaleenType = BooleanType()
+                public valboolean: BaleenType = BooleanType()
```
Comment (kdallmeyer-sr) [inline baleen-poet/src/test/kotlin/com/shoprunner/baleen/poet/BaleenPoetTest.kt:56]: All these changes from `val field`  to `public valfield` look weird

Comment (HawaiianSpork) [reply baleen-poet/src/test/kotlin/com/shoprunner/baleen/poet/BaleenPoetTest.kt:56]: Yeah, the kotlin poet guys seem pretty admit that it should be there https://github.com/square/kotlinpoet/issues/1001

Comment (HawaiianSpork) [reply baleen-poet/src/test/kotlin/com/shoprunner/baleen/poet/BaleenPoetTest.kt:56]: Hmm, sorry.  rereading your comment.  Let me look into that.

Comment (HawaiianSpork) [reply baleen-poet/src/test/kotlin/com/shoprunner/baleen/poet/BaleenPoetTest.kt]: fixed

**Review by kdallmeyer-sr — COMMENTED** [review]
Yes. The spacing was weird. Also weird that they now generate an explicit public

**Pull request merged**